crates/celo-revm/src/handler.rs

```diff
@@ -171,12 +193,43 @@ where
         // Collect logs from the system call to be included in the final receipt
         evm.ctx()
             .chain()
-            .cip64_actual_tx_system_call_logs_post
+            .actual_cip64_tx_info
+            .logs_post
             .extend(logs);
-
+        evm.ctx()
+            .chain()
+            .actual_cip64_tx_info
+            .actual_intrinsic_gas_used += gas_used;
+        self.warn_if_gas_cost_exceeds_intrinsic_gas_cost(evm, fee_currency)?;
         Ok(())
     }
 
+    fn warn_if_gas_cost_exceeds_intrinsic_gas_cost(
+        &self,
+        evm: &mut CeloEvm<DB, INSP>,
```

Comment:
We could pass only `evm.ctx().chain()` instead here.

Reply:
Not anymore with the new refactor, hehe